tasks/package.js

```diff
@@ -13,35 +13,110 @@ module.exports = function(grunt) {
     var path = require('path');
     grunt.loadNpmTasks('grunt-contrib-copy');
 
-    var config = grunt.config.get('config.packages'),
+    var config = grunt.config.get('config.packages.default'),
       srcFiles = ['**', '!**/.gitkeep'].concat((config && config.srcFiles && config.srcFiles.length) ? config.srcFiles : '**'),
       projFiles = (config && config.projFiles && config.projFiles.length) ? config.projFiles : [];
 
     var destPath = grunt.config.get('config.buildPaths.package') + '/package';
     var tasks = []
 
-    grunt.config('copy.package', {
-      files: [
-        {
-          expand: true,
-          cwd: '<%= config.buildPaths.html %>',
-          src: srcFiles,
-          dest: destPath + (grunt.config.get('config.packages.dest.docroot') || ''),
-          dot: true
-        },
-        {
+    // Figure out if we are building a vendor-specific package.
+    var buildTarget = grunt.option('target') || process.env.GRUNT_PACKAGE_TARGET;
+    if (buildTarget && isValidTarget(buildTarget)) {
+      grunt.config('config.buildTarget', grunt.option('target'))
+    }
+    else {
+      grunt.config('config.buildTarget', 'default');
+    }
+    grunt.log.writeln('Building package for ' + grunt.config.get('config.buildTarget'));
+
+    var validTarget = grunt.config.get('config.buildTarget');
+
+    // Build the specified vendor target, or the default.
+    switch (validTarget) {
```

Comment:
Why do we need to custom wrangle Acquia? The only significant difference from the default expectations is the shared directory. As I asked in another comment, not sure the role of that directory but we can follow the pattern in place for dest.docroot and dest.devResources and have a generic loop for all things in package.dest to catch and copy them.